### PR TITLE
Set permission to certs to allow correct removing on win (port to 3)

### PR DESCRIPTION
**Pull request by luipir**
## Description
port to 3.0 of https://github.com/qgis/QGIS/pull/5477 with added unit tests.

ISSUE to resolve: On Windows, PKI PostGIS connections does not remove dumped and temporary certs on temp directory

For PKI db connections (at least Postgis) certs have to be dumped on files to setup connection and then removed.
File permissions are set here: https://qgis.org/api/2.18/qgsauthcertutils_8cpp_source.html#l00271
and python os.remove or QFile::remove(file) are enough on Linux and OSX. But not enough on Win because of ownership difference [1].
So it's necessary to explicitly setup minimum write permission to remove the file. The minimum is
QFile::WriteOwner.
In this moment only PostGIS provider is PKI enabled. The others DB, as far as I can read, still use basic auth.

This PR affect Postgis provider for c++ side and db_manager for the python side.

The issue need to be forward ported to master, but higher priority has the current working release.

[1] http://doc.qt.io/qt-4.8/qfile.html#Permission-enum

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by elpaso** [comment]
Can you please fix the test issues?
- https://travis-ci.org/qgis/QGIS/jobs/300239711#L1916
- https://travis-ci.org/qgis/QGIS/jobs/300239712#L639
- https://travis-ci.org/qgis/QGIS/jobs/300239712#L673

File: python/plugins/db_manager/db_plugins/postgis/connector.py [hunk 1]
```diff
@@ -66,7 +66,8 @@ def __init__(self, uri):
         try:
             self.connection = psycopg2.connect(expandedConnInfo)
         except self.connection_error_types() as e:
-            err = str(e)
+            # get credentials if cached or assking to the user no more than 3 times
```
Comment (elpaso) [inline python/plugins/db_manager/db_plugins/postgis/connector.py]: typo: asking

File: src/providers/postgres/qgspostgresconn.cpp [hunk 2]
```diff
@@ -229,7 +229,26 @@ QgsPostgresConn::QgsPostgresConn( const QString &conninfo, bool readOnly, bool s
     {
       QString fileName = expandedUri.param( param );
       fileName.remove( QStringLiteral( "'" ) );
-      QFile::remove( fileName );
+      QFile file( fileName );
+      // set minimal permission to allow removing on Win.
+      // On linux and Mac if file is set with QFile::>ReadUser
+      // does not create problem removin certs
```
Comment (elpaso) [inline src/providers/postgres/qgspostgresconn.cpp]: removing

File: src/providers/postgres/qgspostgresconn.cpp [hunk 3]
```diff
@@ -229,7 +229,26 @@ QgsPostgresConn::QgsPostgresConn( const QString &conninfo, bool readOnly, bool s
     {
       QString fileName = expandedUri.param( param );
       fileName.remove( QStringLiteral( "'" ) );
-      QFile::remove( fileName );
+      QFile file( fileName );
+      // set minimal permission to allow removing on Win.
+      // On linux and Mac if file is set with QFile::>ReadUser
```
Comment (elpaso) [inline src/providers/postgres/qgspostgresconn.cpp]: extra `>`

**Review by elpaso — CHANGES_REQUESTED** [review]
Please have a look to the failing test issues (see comment in the PR)

**Comment by luipir** [comment]
working to resurrect postgres in the docker test as in https://github.com/qgis/QGIS/pull/5550

**Comment by luipir** [comment]
closed in favour of https://github.com/qgis/QGIS/pull/6647

**Pull request closed without merging**